src/lib/http_server.cpp

```diff
@@ -0,0 +1,84 @@
+#include "http_server.hpp"
+#include "iomgr_config.hpp"
+#include <sisl/auth_manager/security_config.hpp>
+#include <ifaddrs.h>
+#include <arpa/inet.h>
+
+namespace iomgr {
+
+HttpServer::HttpServer() {
+    Pistache::Address addr(Pistache::Ipv4::any(), Pistache::Port(IM_DYNAMIC_CONFIG(io_env.http_port)));
+    m_http_endpoint = std::make_unique< Pistache::Http::Endpoint >(addr);
+    auto flags = Pistache::Tcp::Options::ReuseAddr;
+    auto opts = Pistache::Http::Endpoint::options()
+                    .threadsName("http_server")
+                    .maxRequestSize(IM_DYNAMIC_CONFIG(io_env.http_max_request_size))
+                    .threads(IM_DYNAMIC_CONFIG(io_env.http_num_threads))
+                    .flags(flags);
+    m_http_endpoint->init(opts);
+    setup_ssl();
+    get_local_ips();
+}
+
+void HttpServer::start() {
+    // setup auth middleware
+    m_router.addMiddleware(Pistache::Rest::Routes::middleware(&HttpServer::do_auth, this));
+
+    // setup all routes and start the server
+    m_http_endpoint->setHandler(m_router.handler());
+    m_http_endpoint->serveThreaded();
+}
+
+void HttpServer::setup_route(Pistache::Http::Method method, std::string resource,
+                             Pistache::Rest::Route::Handler handler, url_type const& type) {
+    m_router.addRoute(std::move(method), resource, std::move(handler));
+
```

Comment:
Since setting up the route and accessing is from unordered_set and not a protected list, can you add a check to see if setup_route is called before all init(). You can assert in debug and warn in release.